### PR TITLE
Prepare distribution mainnet

### PR DESCRIPTION
**Pull request by Utilitycoder**
# Protocol Fee Implementation

Adds protocol fee functionality to the distributor contract:
- Adds protocol fee percentage (basis points) and address storage
- Implements fee calculation and collection in both distribution methods
- Updates constructor to accept protocol fee address and owner
- Adds tests for fee calculation and edge cases
- Updates dependencies to latest versions

Technical changes:
- Fee is calculated as percentage of total distributed amount
- Fee is collected in addition to distributed amounts (recipients get full amount)
- Owner can update fee percentage and collection address
- Maximum fee is 100% (10000 basis points)

### CODE REVIEW

File: src/distribute.cairo [hunk 1]
```diff
@@ -167,10 +193,13 @@ mod Distributor {
...
+            assert(!token.is_zero(), INVALID_TOKEN);
 
             let caller = get_caller_address();
+            let contract_address = get_contract_address();
             let token_dispatcher = IERC20Dispatcher { contract_address: token };
-            let mut total_amount: u256 = 0;
+            let mut amount_to_distribute: u256 = 0;
+            let timestamp: u64 = get_block_timestamp();
 
             // Calculate total amount needed
```
Comment (mubarak23) [inline src/distribute.cairo:204]: let change the test here to reflect what the amount is needed for i.e 

` Calculate total amount needed for distribution`

File: src/distribute.cairo [hunk 2]
```diff
@@ -196,21 +235,60 @@ mod Distributor {
...
                 );
         }
 
+        fn get_protocol_fee_percent(self: @ContractState) -> u256 {
+            self.protocol_fee_percent.read()
+        }
+
+        fn set_protocol_fee_percent(ref self: ContractState, new_fee_percent: u256) {
+            self.ownable.assert_only_owner();
+            self.protocol_fee_percent.write(new_fee_percent);
```
Comment (mubarak23) [inline src/distribute.cairo:280]: can we make sure we are not setting zero as protocol fee, or do we want at any point in time in the future to set protocol fee to zero

**Review by mubarak23 — CHANGES_REQUESTED** [review]

**Review by mubarak23 — APPROVED** [review]
